accounts-db/src/tiered_storage/hot.rs

```diff
@@ -316,6 +318,83 @@ impl HotStorageReader {
     }
+
+    /// Returns the size of the account block based on its account offset
+    /// and index offset.
+    ///
+    /// The account block size information is omitted in the hot accounts file
+    /// as it can be derived by comparing the offset of the next hot account
+    /// meta in the index block.
+    fn get_account_block_size(
+        &self,
+        account_offset: HotAccountOffset,
+        index_offset: IndexOffset,
+    ) -> TieredStorageResult<usize> {
+        // the offset that points to the hot account meta.
+        let account_meta_offset = account_offset.offset();
+
+        // Obtain the ending offset of the account block.  If the current
+        // account is the last account, then the ending offset is the
+        // index_block_offset.
+        let account_block_ending_offset =
+            if index_offset.0.saturating_add(1) == self.footer.account_entry_count {
+                self.footer.index_block_offset as usize
+            } else {
+                self.get_account_offset(IndexOffset(index_offset.0.saturating_add(1)))?
+                    .offset()
+            };
+
+        // With the ending offset, minus the starting offset (i.e.,
+        // the account meta offset) and the HotAccountMeta size, the reminder
+        // is the account block size (account data + optional fields).
+        Ok(account_block_ending_offset
+            .saturating_sub(account_meta_offset)
+            .saturating_sub(std::mem::size_of::<HotAccountMeta>()))
+    }
+
+    /// Returns the account block that contains the account associated with
+    /// the specified index given the offset to the account meta and its index.
+    fn get_account_block<'a>(
+        &'a self,
+        account_offset: HotAccountOffset,
+        index_offset: IndexOffset,
+    ) -> TieredStorageResult<&'a [u8]> {
+        let (data, _): (&'a [u8], _) = get_slice(
+            &self.mmap,
+            account_offset.offset() + std::mem::size_of::<HotAccountMeta>(),
+            self.get_account_block_size(account_offset, index_offset)?,
+        )?;
+
+        Ok(data)
+    }
+
+    /// Returns the account located at the specified index offset.
+    pub fn get_account(
+        &self,
+        index_offset: IndexOffset,
+    ) -> TieredStorageResult<Option<(StoredAccountMeta<'_>, usize)>> {
+        if index_offset.0 >= self.footer.account_entry_count {
+            return Ok(None);
+        }
+
+        let account_offset = self.get_account_offset(index_offset)?;
+
+        let meta = self.get_account_meta_from_offset(account_offset)?;
+        let address = self.get_account_address(index_offset)?;
+        let owner = self.get_owner_address(meta.owner_offset())?;
+        let account_block = self.get_account_block(account_offset, index_offset)?;
+
+        Ok(Some((
+            StoredAccountMeta::Hot(TieredReadableAccount {
+                meta,
+                address,
+                owner,
+                index: index_offset.0 as usize,
```

Comment:
I have some thoughts about this actually.  So if accounts db always keep the account address, then here we can also think about returning AccountOffset instead of IndexOffset.  This can save one jump (allow us to directly access the account meta / data without first accessing the index block, then account meta / data).